### DISCUSSION TITLE
Kong nginx add_headers

### DISCUSSION THREAD

**Discussion by fadharpra** [Kong/kong#11826] [Help] [answered]
Hi, I want to set up Nginx to add 'add_header' directives like (Access-Allow-Origin, X-Frame-Options, X-XSS-Protection, X-Content-Type-Options, and Strict-Transport-Security) for ports 8000, 8001, 8443, and 8444.

I'm using Docker to run Kong 3.4.2 and PostgreSQL 13. I can see in /usr/local/kong that there are some files named nginx_kong.conf. I have tried modifying them using the Dockerfile, but it doesn't change anything.

What should I do?

**Comment by samugi** [comment, marked as answer]
@fadharpra please try updating the relevant block of the `nginx_kong.lua` template instead. Alternatively, you should be able to use `KONG_NGINX_HTTP_*` to inject directives [as documented](https://docs.konghq.com/gateway/latest/reference/configuration/#nginx-injected-directives-section).

**Reply by fadharpra** [reply]
I have used KONG_NGINX_HTTP_X-FRAME-OPTIONS: "SAMEORIGIN" in the docker-compose.yml environment but, its still error:

```
2023/11/03 09:17:01 [warn] 1#0: the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /usr/local/kong/nginx.conf:7
nginx: [warn] the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /usr/local/kong/nginx.conf:7
2023/11/03 09:17:01 [error] 1#0: init_by_lua error: /usr/local/share/lua/5.1/kong/init.lua:575: nginx_http_x-frame-options is not a string: 'true'
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/kong/init.lua:575: in function 'init'
        init_by_lua:3: in main chunk
nginx: [error] init_by_lua error: /usr/local/share/lua/5.1/kong/init.lua:575: nginx_http_x-frame-options is not a string: 'true'
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/kong/init.lua:575: in function 'init'
        init_by_lua:3: in main chunk
Error: nginx_http_x-frame-options is not a string: 'true'

  Run with --v (verbose) or --vv (debug) for more details
Error: nginx_http_x-frame-options is not a string: 'true'

```

**Reply by samugi** [reply]
I think the syntax is incorrect, please try `KONG_NGINX_HTTP_ADD_HEADER: "X-Frame-Options SAMEORIGIN"`
